bundles/io/org.eclipse.smarthome.io.security/src/main/java/org/eclipse/smarthome/io/security/api/Cryptographic.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.security.api;
+
+import java.security.GeneralSecurityException;
+
+/**
+ * Implement this in {@link TlsNetworkServer} to get notifications of changed
+ * certificate contexts.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public interface Cryptographic {
+    /**
+     * Encrypts the given text by the given password.
+     *
+     * @param password The password as byte array.
+     * @param plain The plain text to encrypt
+     * @return Returns the ciphered text.
+     * @throws GeneralSecurityException
+     */
+    char[] encrypt(byte[] password, char[] plain) throws GeneralSecurityException;
+
+    /**
+     * Decrypts the given text with the given password.
+     *
+     * @param password The password as byte array.
+     * @param cipher The cipher to decrypt
+     * @return Returns the plain text
+     * @throws GeneralSecurityException
+     */
+    char[] decrypt(byte[] password, char[] cipher) throws GeneralSecurityException;
+
+    /**
+     * Encrypts the given bytes by the given password.
+     *
+     * @param password The password as byte array.
+     * @param plain The plain bytes to encrypt
+     * @return Returns the ciphered text.
+     * @throws GeneralSecurityException
+     */
+    byte[] encrypt(byte[] password, byte[] plain) throws GeneralSecurityException;
```

Comment:
In the `SecretsVaultService` you are working on `ByteBuffer`s and here you are using `byte[]` if the `ByteBuffer` is really more convenient maybe we should use it here too, or change both to use `byte[]` to have it consistent.

Reply:
I'm using ByteBuffer if the argument can be null and byte arrays otherwise.

Reply:
If there is a param defined as `byte[] param` one can also pass `null`, so I am confused about your explanation.

Reply:
I can't use annotations on a byte array, at least it didn't work for me. I have no problems of using ByteBuffer all the time. It just wraps a byte array and has almost no additional runtime costs.

Reply:
If it doesn't add "much" (I don't think "no") additional runtime cost we could use it. But I wouldn't make this decision dependent on whether you can add a null annotation or not. Otherwise you also have to replace all "int", "boolean", etc. parameters in the code as well because you cannot add the null annotations on primitives.

Reply:
Actually I do this. For the Mqtt connection object for example. The port parameter is an Integer instead of an int so that the user can set the parameter explicitly to null if he doesn't care. For booleans it makes sense too sometimes if you need a tristate (on/off/disabled).